flink-java/src/main/java/org/apache/flink/api/java/utils/MultipleParameterTool.java

```diff
@@ -0,0 +1,269 @@
+			Objects.equals(defaultData, that.defaultData) &&
+			Objects.equals(unrequestedParameters, that.unrequestedParameters);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(data, defaultData, unrequestedParameters);
+	}
+
+	// ------------------ Get data from the util ----------------
+
+	/**
+	 * Returns number of parameters in {@link ParameterTool}.
+	 */
+	@Override
+	public int getNumberOfParameters() {
+		return data.size();
+	}
+
+	/**
+	 * Returns the String value for the given key. The value should only have one item.
+	 * Use {@link #getMultiParameter(String)} instead if want to get multiple values parameter.
+	 * If the key does not exist it will return null.
+	 */
+	@Override
+	public String get(String key) {
+		addToDefaults(key, null);
+		unrequestedParameters.remove(key);
+		if (!data.containsKey(key)) {
+			return null;
+		}
+		Preconditions.checkState(data.get(key).size() == 1,
+			"Key %s should has only one value.", key);
+		return (String) data.get(key).toArray()[0];
+	}
+
+	/**
+	 * Check if value is set.
+	 */
+	@Override
+	public boolean has(String value) {
+		addToDefaults(value, null);
+		unrequestedParameters.remove(value);
+		return data.containsKey(value);
+	}
+
+	/**
+	 * Returns the Collection of String values for the given key.
+	 * If the key does not exist it will return null.
+	 */
+	public Collection<String> getMultiParameter(String key) {
+		addToDefaults(key, null);
+		unrequestedParameters.remove(key);
+		return data.getOrDefault(key, null);
+	}
+
+	/**
+	 * Returns the Collection of String values for the given key.
+	 * If the key does not exist it will throw a {@link RuntimeException}.
+	 */
+	public Collection<String> getMultiParameterRequired(String key) {
+		addToDefaults(key, null);
+		Collection<String> value = getMultiParameter(key);
+		if (value == null) {
+			throw new RuntimeException("No data for required key '" + key + "'");
+		}
+		return value;
+	}
+
+	// ------------------------- Export to different targets -------------------------
+
+	/**
+	 * Return MultiMap of all the parameters processed by {@link MultipleParameterTool}.
+	 *
+	 * @return MultiMap of the {@link MultipleParameterTool}. Key is String and Value is a Collection of String.
+	 */
+	public Map<String, Collection<String>> toMultiMap() {
+		return data;
+	}
+
+	@Override
+	protected Object clone() throws CloneNotSupportedException {
+		return new MultipleParameterTool(this.data);
+	}
+
+	// ------------------------- Interaction with other ParameterUtils -------------------------
+
+	/**
+	 * Merges two {@link MultipleParameterTool}.
+	 *
+	 * @param other Other {@link MultipleParameterTool} object
+	 * @return The Merged {@link MultipleParameterTool}
+	 */
+	public MultipleParameterTool mergeWith(MultipleParameterTool other) {
+		final Map<String, Collection<String>> resultData = new HashMap<>(data.size() + other.data.size());
+		resultData.putAll(data);
+		resultData.putAll(other.data);
```

Comment:
what if a key is present both in `this` and `other`? Should we join the collections? 

Frankly If you don't want to spend time fixing it, I would be fine with dropping this method as we do not need it for now. It always can be added in the future. Of course if you would prefer to keep it, I'm also fine with it.

Reply:
Yes, i think we should join the collection. I will fix it quickly.